src/uu/cp/src/cp.rs

```diff
@@ -1910,6 +1930,35 @@ fn copy_helper(
     Ok(())
 }
 
+/// Copy the file from `source` to `dest` by reading into and writing from a buffer.
+/// Increment the progress bar on each write to show the intermediate progress.
+pub fn copy_with_intermediate_progress(
+    source: &Path,
+    dest: &Path,
+    progress_bar: &Option<ProgressBar>,
+) {
+    let metadata = fs::metadata(source).unwrap();
```

Comment:
I think you should handle the error and not use unwrap() for this line and the next

Reply:
Done!